common-pg/src/main/scala/com/socrata/pg/store/PGSecondaryDatasetMapReader.scala

```diff
@@ -1,53 +1,167 @@
 package com.socrata.pg.store
 
-import java.sql.Connection
 import com.rojoma.simplearm.v2._
-import com.socrata.datacoordinator.id.{DatasetId, RollupName}
+import com.socrata.datacoordinator.id.{CopyId, DatasetId, RollupName}
+import com.socrata.datacoordinator.truth.metadata.sql.{BasePostgresDatasetMapReader, PostgresDatasetMapReader}
+import com.socrata.datacoordinator.truth.metadata.{CopyInfo, DatasetInfo, LifecycleStage, TypeNamespace}
 import com.socrata.datacoordinator.util.TimingReport
-import com.socrata.datacoordinator.truth.metadata.sql.{PostgresDatasetMapReader, BasePostgresDatasetMapReader}
-import com.socrata.datacoordinator.truth.metadata.{TypeNamespace, CopyInfo}
-import com.socrata.soql.types.SoQLType
+import org.joda.time.DateTime
+
+import java.sql.Connection
+import scala.reflect.ClassTag
 
-trait LocalRollupReaderOverride[CT] { self: BasePostgresDatasetMapReader[CT] =>
-  def localRollupsQuery = "SELECT name, soql, table_name FROM rollup_map WHERE copy_system_id = ?"
+trait LocalRollupReaderOverride[CT] {
+  self: BasePostgresDatasetMapReader[CT] =>
   override def rollups(copyInfo: CopyInfo): Seq[LocalRollupInfo] = {
     using(conn.prepareStatement(localRollupsQuery)) { stmt =>
       stmt.setLong(1, copyInfo.systemId.underlying)
       using(t("rollups", "copy_id" -> copyInfo.systemId)(stmt.executeQuery())) { rs =>
         val res = Vector.newBuilder[LocalRollupInfo]
-        while(rs.next()) {
-          res += new LocalRollupInfo(copyInfo, new RollupName(rs.getString("name")), rs.getString("soql"), rs.getString("table_name"))
+        while (rs.next()) {
+          res += new LocalRollupInfo(copyInfo, new RollupName(rs.getString("name")), rs.getString("soql"), rs.getString("table_name"), RollupId(rs.getLong("system_id")))
         }
         res.result()
       }
     }
   }
 
-  def localRollupQuery = "SELECT soql, table_name FROM rollup_map WHERE copy_system_id = ? AND name = ?"
+  def localRollupsQuery = "SELECT system_id, name, soql, table_name FROM rollup_map WHERE copy_system_id = ?"
+
   override def rollup(copyInfo: CopyInfo, name: RollupName): Option[LocalRollupInfo] = {
     using(conn.prepareStatement(localRollupQuery)) { stmt =>
       stmt.setLong(1, copyInfo.systemId.underlying)
       stmt.setString(2, name.underlying)
-      using(t("rollup", "copy_id" -> copyInfo.systemId,"name" -> name)(stmt.executeQuery())) { rs =>
-        if(rs.next()) {
-          Some(new LocalRollupInfo(copyInfo, name, rs.getString("soql"), rs.getString("table_name")))
+      using(t("rollup", "copy_id" -> copyInfo.systemId, "name" -> name)(stmt.executeQuery())) { rs =>
+        if (rs.next()) {
+          Some(new LocalRollupInfo(copyInfo, name, rs.getString("soql"), rs.getString("table_name"), RollupId(rs.getLong("system_id"))))
         } else {
           None
         }
       }
     }
   }
+
+  def localRollupQuery = "SELECT system_id, soql, table_name FROM rollup_map WHERE copy_system_id = ? AND name = ?"
```

Comment:
can we move this up to where localRollupsQuery (note the plural) so it's obvious this query is being executed and not the localRollupsQuery?